src/Futhark/Optimise/GenRedOpt.hs

```diff
@@ -105,7 +105,8 @@ genRed2Tile2d env kerstm@(Let pat_ker aux (Op (SegOp (SegMap seg_thd seg_space k
     -- some `code1`, followed by one accumulation, followed by some `code2`
     -- UpdateAcc VName [SubExp] [SubExp]
     (code1, Just accum_stmt, code2) <- matchCodeAccumCode kstms,
-    Let pat_accum _aux_acc (BasicOp (UpdateAcc acc_nm acc_inds acc_vals)) <- accum_stmt,
+    -- TODO: should Unsafe UpdateAccs be treated differently?
```

Comment:
Ask Cosmin, but I think not.

Reply:
I would also think not. @coancea do you have an opinion?